### ISSUE TITLE
Obtain `ProjectRequestToDescriptionConverter` from the application context

### ISSUE THREAD

**Issue by mmoayyed** [spring-io/initializr#1209]
Using `0.10.1`.

Given the following bean:

```java
		@Bean
		@ConditionalOnMissingBean
		ProjectGenerationController<ProjectRequest> projectGenerationController(
				InitializrMetadataProvider metadataProvider,
				ObjectProvider<ProjectRequestPlatformVersionTransformer> platformVersionTransformer,
				ApplicationContext applicationContext) {
			ProjectGenerationInvoker<ProjectRequest> projectGenerationInvoker = new ProjectGenerationInvoker<>(
					applicationContext, new DefaultProjectRequestToDescriptionConverter(platformVersionTransformer
							.getIfAvailable(DefaultProjectRequestPlatformVersionTransformer::new)));
			return new DefaultProjectGenerationController(metadataProvider, projectGenerationInvoker);
		}
```

The `ProjectGenerationInvoker` component uses a *hardcoded* reference to `DefaultProjectRequestToDescriptionConverter`, which makes it somewhat difficult to supply injected implementations of `ProjectRequestToDescriptionConverter`. 

Obtaining a reference to `ProjectRequestToDescriptionConverter` could be done using an `ObjectProvider` from the application context. If none is found, the default implementation could be used. Alternatively, one can of course design a replacement for `ProjectGenerationController`, which seems rather overkill.

Happy to submit a pull request and contribute this change, if deemed reasonable.

**Comment by snicoll** [comment]
Thanks for the suggestion but I don't see a lot of value in doing that. 

Please consider the generic on `ProjectRequest`. Using a specific converter is primarily meant to use cases where the `ProjectRequest` has extra attributes (and therefore need a dedicated extension of `ProjectRequest`). This is also covered [in the reference guide](https://docs.spring.io/initializr/docs/current/reference/html/#create-instance-advanced-config-custom-project-request).

**Comment by mmoayyed** [comment]
Appreciate the reference. I will follow up with the approach pointed out in the guide.

PS Small typo in that section by the way. It reads, *A custom instance may chose however to...*. Should be *choose*.

**Comment by snicoll** [comment]
> PS Small typo in that section by the way. It reads, A custom instance may chose however to.... Should be choose.

Cheers. I've fixed that in 73d780e1cd74d5a0d181b732a61c15ecea7780cb
